server/src/routines/game.ts

```diff
@@ -338,17 +338,20 @@ function checkHermitHealth(game: GameModel) {
 			if (card.props.category === 'hermit') {
 				playerState.lives -= 1
 
-				// reward card
+				// prize card
 				if (game.settings.disableRewardCards) continue
-				game.components
+				const prizeCard = game.components
 					.filter(
 						CardComponent,
 						query.card.slot(query.slot.deck),
 						query.card.player(playerState.entity),
 					)
 					.sort(CardComponent.compareOrder)
 					.at(0)
-					?.draw(playerState.opponentPlayer.entity)
+				if (prizeCard) {
+					prizeCard.draw(playerState.opponentPlayer.entity)
```

Comment:
I think this should be the other way round, so that when onChangeSlot is called it's marked as a prizeCard already.